src/modules/app/browser-init.service.ts

```diff
@@ -134,4 +137,23 @@ export class BrowserInitService extends InitService {
   protected initGoogleAnalytics() {
     this.googleAnalyticsService.addTrackingIdToPage();
   }
+
+  /**
+   * During the external authentication flow invalidates the SSR transferState
+   * data in the cache. This allows the app to fetch fresh content.
+   * @private
+   */
+  private externalAuthCheck() {
+
+    this.authService.isExternalAuthentication().pipe(
+        filter((externalAuth: boolean) => externalAuth)
+      ).subscribe(() => {
+        // Clear the transferState data.
+        this.rootDatatService.invalidateRootCache();
+        this.authService.setExternalAuthStatus(false);
+      }
```

Comment:
In other places, if we want to make sure we clear the cache (e.g. when logging out, or when changing the language), [we force the page to reload](https://github.com/DSpace/dspace-angular/blob/4403555c29f8b0a48dfb3671f85b7b93cd017d1b/src/app/core/auth/auth.service.ts#L471-L480). Perhaps that's more fool proof and less resource intensive, than setting _all_ requests to stale in one go?

Reply:
Yes, I initially used a hard redirect (with a notification that told the user that they were successfully authenticated and we were fetching the content). It is fool proof and it works. But the user experience is not good at all whereas this approach is fairly quick and seamless for the user.